src/features/view-nft/NFTDetailsCard/ui/Options/Options.tsx

```diff
@@ -1,53 +1,129 @@
- */
 export const Options: FC<OptionsProps> = ({ zna }) => {
 	const domainId = getDomainId(zna);
 
 	const { account } = useWeb3();
+	const { shareAsset } = useShareAsset(zna);
+	const { downloadAsset } = useDownloadAsset();
 	const { data: domain } = useDomainData(domainId);
+	const { data: metadata } = useDomainMetadata(domainId);
 
-	return (
-		<div className={styles.Container}>
-			{/* TODO: wire up share to twitter */}
-			<Tooltip content="Share to Twitter">
-				<button onClick={() => console.log('Share')}>
-					<IconShare7 color={'#52CBFF'} isFilled />
-				</button>
-			</Tooltip>
+	const downloadOptions = getDownloadOptions(
+		downloadAsset,
+		metadata?.image,
+		metadata?.image_2,
+		metadata?.image_3,
+		metadata?.animation_url,
+	);
 
-			{/* TODO: wire up download for twitter */}
-			<Tooltip content="Download for Twitter">
-				<button onClick={() => console.log('Download')}>
-					<IconDownload2 color={'#52CBFF'} isFilled />
-				</button>
-			</Tooltip>
+	const onShareAsset = () => {
+		shareAsset(NFT_ASSET_SHARE_KEYS.TWITTER);
+	};
 
+	return (
+		<div className={styles.Options}>
+			<ShareButton onClick={onShareAsset} />
+			<DownloadButton downloadOptions={downloadOptions} />
 			{domain?.owner?.toLowerCase() === account?.toLowerCase() && (
-				<MoreNFTOptions
+				<MoreOptionsButton
+					className={styles.Button}
 					zna={zna}
 					trigger={<IconDotsVertical className={styles.Icon} isFilled />}
 				/>
 			)}
 		</div>
 	);
 };
+
+/********************
+ * Share Asset Button
+ ********************/
+const ShareButton = ({ onClick }) => (
+	<Tooltip content="Share to Twitter">
+		<button className={styles.Button} onClick={onClick}>
+			<IconShare7 color={'#52CBFF'} isFilled />
+		</button>
+	</Tooltip>
+);
+
+/***********************
+ * Download Asset Button
+ ***********************/
+const DownloadButton = ({ downloadOptions }) => (
+	<DropdownMenu
+		className={styles.Button}
+		items={downloadOptions}
+		side="bottom"
+		alignMenu="end"
+		trigger={
+			<Tooltip content="Download for Twitter">
+				<IconDownload2 color={'#52CBFF'} isFilled />
+			</Tooltip>
+		}
+	/>
+);
+
+/*******************************
+ * More Options Dropdown Button
+ ******************************/
+export const MoreOptionsButton = ({ zna, trigger, className }) => {
+	const [option, setOption] = useState<DropdownOption | undefined>();
+
+	const onSelectOption = useCallback((e) => {
+		setOption(e.id);
+	}, []);
+
+	return (
+		<>
+			{dropdownOptionConfig.map(({ id, modalComponent: Modal }) => (
+				<Modal
+					key={id}
+					zna={zna}
+					open={option === id}
+					onOpenChange={(open) => !open && setOption(undefined)}
+					onClose={() => setOption(undefined)}
+				/>
+			))}
+
+			<DropdownMenu
+				className={className}
+				items={dropdownOptionConfig.map(({ id, label, icon }) =>
+					generateDropdownOption(id, label, icon, onSelectOption),
```

Comment:
nit: remove the `generateDropdownOption` function and do 

```
items={dropdownOptionConfig.map((params) =>
					({ ...params, onSelectOption }),
}
```

Not sure if that will work exactly, but should use that kind of syntax. I can understand wrapping this in a function, but imo should just handle it inline as the function isn't handling any complex logic.